### PR TITLE
[ShapeOpt] Adding stopping criteria in trust region algorith by relative tolerance

### PR DESCRIPTION
**Pull request by marcnunezc**
Hi!

Current trust region algorithm does not have an stopping criteria (apart from the maximum iterations limit). Is there an specific reason for this related to the algorithm or is it something that was simply not done yet?

In this PR I'm adding an stopping criteria with the relative tolerance, using the same function as the steepest descent algorithm to check if the algorithm is converged.

### CODE REVIEW

File: applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py [hunk 1]
```diff
@@ -129,11 +130,31 @@ def RunOptimizationLoop(self):
...
     def FinalizeOptimizationLoop(self):
         self.analyzer.FinalizeAfterOptimizationLoop()
         self.data_logger.FinalizeDataLogging()
 
+    # --------------------------------------------------------------------------
+    def __isAlgorithmConverged(self):
+
+        if self.opt_iteration > 1 :
+            # Check if maximum iterations were reached
+            if self.opt_iteration == self.algorithm_settings["max_iterations"].GetInt():
```
Comment (armingeiser) [inline applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py:146]: if this is tested here, the loop above could be
```python
while not self.__isAlgorithmConverged():
```

File: applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py [hunk 2]
```diff
@@ -129,11 +130,31 @@ def RunOptimizationLoop(self):
...
+                KM.Logger.Print("")
+                KM.Logger.PrintInfo("ShapeOpt", "Maximal iterations of optimization problem reached!")
+                return True
+
+            # Check for relative tolerance
+            relativeChangeOfObjectiveValue = self.data_logger.GetValues("rel_change_objective")[self.opt_iteration]
```
Comment (armingeiser) [inline applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py]: relative_change_of_objective_value - local variables in lower case

File: applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py (hunk 1 above)
Comment (armingeiser) [reply applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py:146]: Forget it then we would need to increment the counter, i prefer the way you did it.

**Review by armingeiser — CHANGES_REQUESTED** [review]
looks good for me, only a small naming change

File: applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py (hunk 2 above)
Comment (marcnunezc) [reply applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py]: Ok! I went ahead and updated that in the rest of algorithms for consistency

**Review by armingeiser — APPROVED** [review]
Thanks!

**Pull request merged**